### ISSUE TITLE
Unknown error in injected C++ completion routine.

### ISSUE THREAD

**Issue by DDoSolitary** [DDoSolitary/LxRunOffline#1]
@aslandevops:
> After command LxRunOffline /install I have this error
> Error: Failed to launch LxRun. System.ApplicationException: STATUS_INTERNAL_ERROR: Unknown error in injected C++ completion routine. (Code: 15) at EasyHook.RemoteHooking.CreateAndInject(String InEXEPath, String InCommandLine, Int32 InProcessCreationFlags, InjectionOptions InOptions, String InLibraryPath_x86, String InLibraryPath_x64, Int32& OutProcessId, Object[] InPassThruArgs) at LxRunOffline.Program.Main(String[] args)

**Comment by DDoSolitary** [comment]
@aslandevops Could you please try to run `lxrun.exe` directly in command line?

**Comment by aslandevops** [comment]
@DDoSolitary after lxrun.exe i have this:
`Performs administrative operations on the LX subsystem

Usage:
    /install - Installs the subsystem
        Optional arguments:
            /y - Do not prompt user to accept
    /uninstall - Uninstalls the subsystem
        Optional arguments:
            /full - Perform a full uninstall
            /y - Do not prompt user to accept
    /setdefaultuser - Configures the subsystem user that bash will be launched as. If the user does not exist it will be created.
        Optional arguments:
            username - Supply the username
            /y - If username is supplied, do not prompt to create a password
    /update - Updates the subsystem's package index`

**Comment by DDoSolitary** [comment]
And what's your OS version?

**Comment by aslandevops** [comment]
Windows 10 (v1607) 64bit

**Comment by DDoSolitary** [comment]
Any antivirus software installed? As this project uses hooking to change the behavior of `lxrun.exe`, so it may be considered a virus.

**Comment by aslandevops** [comment]
system is clean from any applications (and antivirus to), just installed.

**Comment by DDoSolitary** [comment]
It's hard to say what is wrong now. And I can't reproduce this issue with my PC(Win10 Enterprise Insider Preview Build 15048). Maybe I have to create a virtual machine to test this. But I have a lot of things to do this weekend, and since I'm still a student, I don't have time for this on weekdays. So it seems that I can't start to try to fix this problem before next weekend. Sorry. 😢

**Comment by aslandevops** [comment]
Ok. Thanks a lot!

**Comment by DDoSolitary** [comment]
@aslandevops I've tested my project on an environment same as your described(win10 v1607 x64), but it works well...
I'm sorry but I can't help you as I can't reproduce this issue.

**Comment by secure-code** [comment]
Same error I am also getting. My OS edition is Win10 Enterprise build 1607

**Comment by hillin** [comment]
To me the `LxRun.exe` does not exist in `C:\Windows\System32`, but I can find it in `C:\Windows\WinSxS\amd64_microsoft-windows-lxss-installer_31bf3856ad364e35_10.0.15063.0_none_1525b04a87b3edd7`. However, changing the path in the source code does not fix this problem, `LxRun.exe` just crashes directly.

**Comment by DDoSolitary** [comment]
@hillin This is because you didn't enable the `Windows Subsystem for Linux` feature. Here is the installation guide provided by Microsoft: https://msdn.microsoft.com/en-us/commandline/wsl/install_guide

**Comment by DDoSolitary** [comment]
@secure-code Could you please confirm that the error message you got is the same as @aslandevops? If so, I can close #2 and focus on this issue.

**Comment by hillin** [comment]
@DDoSolitary I checked again (with my Windows Subsystem for Linux eventually successfully installed in the official way) and yes, the file is there. Turns out it's Everything (a file search tool which I always trusted...) not having that file indexed. Thanks!

**Comment by DDoSolitary** [comment]
@secure-code @aslandevops 
I've tested this project on VMs with Win10 x64 Pro v1607 and v1703, but they all worked without any problems.
Could you please provide more detailed steps on how to reproduce this problem?

**Comment by lorinkoz** [comment]
Same issue here
Windows 10 Pro (Creators Update)

using: 16.4.1-server-cloudimg-amd64-root.tar.gz
antivirus disabled

```
Error: Failed to launch LxRun.
System.ApplicationException: STATUS_INTERNAL_ERROR: Unknown error in injected C++ completion routine. (Code: 15)
   en EasyHook.RemoteHooking.CreateAndInject(String InEXEPath, String InCommandLine, Int32 InProcessCreationFlags, InjectionOptions InOptions, String InLibraryPath_x86, String InLibraryPath_x64, Int32& OutProcessId, Object[] InPassThruArgs)
   en LxRunOffline.Program.Main(String[] args)
```

**Comment by DDoSolitary** [comment]
Please try this: http://stackoverflow.com/a/8268636/323899

**Comment by FlutterRage** [comment]
> Please try this: http://stackoverflow.com/a/8268636/323899
Just had the same problem. This did solve it. I had to unblock all files and now it's working. Thanks a lot.

**Comment by DDoSolitary** [comment]
@aslandevops @secure-code @lorinkoz Could you please provide some feedback?

**Comment by DDoSolitary** [comment]
I'm closing this issue according to @FlutterRage's feedback. If you still encounter this problem even after unblocking the files, feel free to reopen this issue or open a new issue.
I'll add notes about unblocking the files in the README later.
